elastic/kibana
# [Connectors] Capturing validation errors in action execution event log

Resolves https://github.com/elastic/kibana/issues/130984

## Summary

Previously, action validation was occurring before event log entries were initialized in the action executor. This meant that if config/secrets/params validation failed, no event log entries would be written for the action execution. Moved the validation step to occur after event log initialization. This does mean that the duration that's captured in the action execution event log now includes the time it takes perform validation, whereas before it strictly captured the duration of the action type executor.

## To verify

* Recreate the action validation error as described in the original issue
* Verify that action event log documents are written including the validation error message
* Verify that the action error is surfaced on the rule details page under the `Error log` tab.

### Checklist
